File: Luogu/P1551/P1551.cpp
#include <iostream>
#include <cstdio>
#define MAXN 5050
using namespace std;
int n, m, p;
int fa[MAXN];

int find (int num)
{
    if (fa[num] != num)
        fa[num] = find (fa[num]);
    return fa[num];
}

int main()
{
    scanf ("%d%d%d", &n, &m, &p);
    for (int i = 1; i <= n; i++)
        fa[i] = i;
    int a, b;
    for (int i = 1; i <= m; i++)
    {
        scanf ("%d%d", &a, &b);
        fa[find(a)] = find(b);
    }
    int x, y;
    while (p--)
    {
        scanf ("%d%d", &x, &y);
        if (find(x) == find(y))
            printf("Yes\n");
        else
            printf("No\n");
    }
    return 0;
}

File: Luogu/P1551/P1551.py
MAXN = 5050
fa = list (range(MAXN))
def find (x):
    if fa[x] != x:
        fa[x] = find (fa[x])
    return fa[x]

n, m, p = map (int, input().split())
for i in range (1, n + 1):
    fa[i] = i
for i in range (1, m + 1):
    a, b = map (int, input().split())
    fa[find(a)] = find(b)
while p:
    x, y = map (int, input().split())
    if find(x) == find(y):
        print ("Yes")
    else:
        print ("No")
    p -= 1